test/collection_test.js

```diff
@@ -64,6 +64,21 @@ describe("Collection", () => {
     };
   }
 
+  function createKeyListIdSchema() {
+    return {
+      generate(record) {
+        let id = [];
+        for (let key in record) {
+          id.push(key);
+        }
+        return id.join(",");
+      },
+      validate(id) {
+        return id !== null && Array.isArray(id.split(","));
```

Comment:
Several things here:

`id != null` would check for both `null` and `undefined`, which may be a worthy addition.

Also a split string will always be an array (`"".split(",") // [""]`) so you may want to update this check as well. 

How about just checking for a non-empty string?